crates/vm/src/expr.rs

```diff
@@ -273,57 +274,53 @@ impl From<Query> for Option<ColumnOp> {
     }
 }
 
-#[derive(Debug, PartialEq, Eq, Clone, Default)]
-#[repr(transparent)]
-/// A set of [`MemTable`]s referenced by a query plan by their [`SourceId`].
+/// A set of [`SourceTable`]s referenced by a query plan by their [`SourceId`].
 ///
-/// Rather than embedding [`MemTable`]s in query plans, we store a `SourceExpr::MemTable`,
+/// Rather than embedding tables in query plans, we store a `SourceExpr::MemTable`,
 /// which contains the information necessary for optimization along with a [`SourceId`].
 /// Query execution then executes the plan, and when it encounters a `SourceExpr::MemTable`,
-/// retrieves the [`MemTable`] from the corresponding `SourceSet`.
+/// retrieves the `T` table from the corresponding `SourceSet`.
 /// This allows query plans to be re-used, though each execution requires a new `SourceSet`.
 ///
-/// Internally, the `SourceSet` stores an `Option<MemTable>` for each planned [`SourceId`].
+/// Internally, the `SourceSet` stores an `Option<T>` for each planned [`SourceId`].
```

Comment:
This comment needs to be revised to describe sensible choices for `T`, and what parts of our codebase use those values. In short, I want the comment to justify why this is generic over `T` rather than simply saying `MemTable`; the current comment is written in a way that makes me go, "What could a `SourceExpr::MemTable` ever refer to other than a `MemTable`?"